modules/azure/src/main/java/org/testcontainers/azure/AzureEventHubsEmulatorContainer.java

```diff
@@ -0,0 +1,138 @@
+    );
+
+    private AzuriteContainer azuriteContainer;
+
+    private boolean useKafka;
+
+    /**
+     * @param dockerImageName specified docker image name to run
+     */
+    public AzureEventHubsEmulatorContainer(final String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    /**
+     * @param dockerImageName specified docker image name to run
+     */
+    public AzureEventHubsEmulatorContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        waitingFor(Wait.forLogMessage(".*Emulator Service is Successfully Up!.*", 1));
+        withExposedPorts(DEFAULT_AMQP_PORT);
+    }
+
+    /**
+     * * Sets the Azurite dependency needed by the Event Hubs Container,
+     *
+     * @param azuriteContainer The Azurite container used by Event HUbs as a dependency
+     * @return this
+     */
+    public AzureEventHubsEmulatorContainer withAzuriteContainer(final AzuriteContainer azuriteContainer) {
+        this.azuriteContainer = azuriteContainer;
+        dependsOn(this.azuriteContainer);
+        return this;
+    }
+
+    /**
+     * Provide the broker configuration to the container.
+     *
+     * @param config The file containing the broker configuration
+     * @return this
+     */
+    public AzureEventHubsEmulatorContainer withConfig(final Transferable config) {
+        withCopyToContainer(config, "/Eventhubs_Emulator/ConfigFiles/Config.json");
+        return this;
+    }
+
+    /**
+     * Accepts the EULA of the container.
+     *
+     * @return this
+     */
+    public AzureEventHubsEmulatorContainer acceptLicense() {
+        return withEnv("ACCEPT_EULA", "Y");
+    }
+
+    /**
+     * Enables Kafka support.
+     *
+     * @return this
+     */
+    public AzureEventHubsEmulatorContainer enableKafka() {
+        this.useKafka = true;
+        return this;
+    }
+
+    @Override
+    protected void configure() {
+        if (azuriteContainer == null) {
+            throw new IllegalStateException(
+                "The image " +
+                getDockerImageName() +
+                " requires an Azurite container. Please provide one with the withAzuriteContainer method!"
+            );
+        }
+        final String azuriteHost = azuriteContainer.getNetworkAliases().get(0);
+        withEnv("BLOB_SERVER", azuriteHost);
+        withEnv("METADATA_SERVER", azuriteHost);
+        // If license was not accepted programmatically, check if it was accepted via resource file
+        if (!getEnvMap().containsKey("ACCEPT_EULA")) {
+            LicenseAcceptance.assertLicenseAccepted(this.getDockerImageName());
+            acceptLicense();
+        }
+        if (this.useKafka) {
+            //Kafka must expose with the fixed default port or the broker's advertised port won't match
+            this.addFixedExposedPort(DEFAULT_KAFKA_PORT, DEFAULT_KAFKA_PORT);
```

Comment:
We must avoid fixed ports. `ConfluentKafkaContainer` and `KafkaContainer` configure the advertised listeners in order to accept connections from host. We should do something similar here if possible. Otherwise, let's remove it for now.

Reply:
Okay, removed Kafka from this container. Most probably anyone using Kafka would use a real Kafka container locally anyways.